nft/nfts.yaml

```diff
@@ -601,6 +601,49 @@ paths:
                       type: number
                       description: 'Floating point value from 0 to 1 representing the prevalence of this value for this trait type.'
       operationId: computeRarity
+  '/{apiKey}/searchContractMetadata':
+    get:
+      summary: searchContractMetadata
+      description: Search for a keyword across metadata of all ERC-721 and ERC-1155 smart contracts
+      tags: ['NFT API Endpoints']
+      parameters:
+        - $ref: '#/components/schemas/apiKey'
+        - $ref: '#/components/schemas/query'
+      x-readme:
+        samples-languages:
+          - javascript
+          - curl
+          - python
+          - go
+      responses:
+        '200':
+          description: 'Returns the list of NFT contracts where the metadata overlaps with the search string.'
+          content:
+            application/json:
+              schema:
+                type: array                  
+                description: 'List of contracts where the metadata overlaps with the search string.'
+                items:
+                  type: object
+                  properties:
+                    address:
+                      $ref: '#/components/schemas/contractAddress'
+                    contractMetadata:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                          description: 'String - The name of the NFT collection'
+                        symbol:
+                          type: string
+                          description: 'String - The symbol of the NFT collection'
+                        totalSupply:
+                          type: string
+                          description: 'String - The total supply for the NFT collection'
+                        tokenType:
+                          type: string
+                          description: 'String - The type of the token (ERC721 or ERC115)'                       
+      operationId: searchContractMetadata
```

Comment:
@ankg since contractMetadata is used both here and in the getContractMetadata response we should create a components/schema for it and use that instead of duplicating definitions for both methods. Right now, the definitions for the same variables are different so this would help bridge consistency there. 

Let me know if that makes sense!